src/predicate/isNaN.ts

```diff
@@ -0,0 +1,15 @@
+/**
+ * Checks if the value is NaN.
+ *
+ * @param {unknown} value - The value to check.
+ * @returns {value is typeof NaN} `true` if the value is NaN, `false` otherwise.
+ *
+ * @example
+ * isNaN(NaN); // true
+ * isNaN(0); // false
+ * isNaN('NaN'); // false
+ * isNaN(undefined); // false
+ */
+export function isNaN(value: unknown): value is typeof NaN {
+  return typeof value === 'number' && value !== value;
```

Comment:
What about we use `Number.isNaN` in this case? We should check the performance differences, though..

Reply:
I checked and found that Number.isNaN has better performance. 
I think it would be better to implement it the way you said.

I think I forgot to use the native function as I focused on utilizing the properties of NaN. 
Thank you for pointing this out!😀
<img width="852" alt="image" src="https://github.com/user-attachments/assets/f21d2503-7b6b-49c5-a69b-435ce26bbd5f">